ocaml/sdk-gen/java/autogen/xen-api/src/main/java/com/xensource/xenapi/CustomDateDeserializer.java

```diff
@@ -45,14 +45,26 @@
 public class CustomDateDeserializer extends StdDeserializer<Date> {
 
     /**
-     * Array of {@link SimpleDateFormat} objects representing the custom date formats
-     * used in XenServer API responses.
+     * Array of {@link SimpleDateFormat} objects representing the date formats
+     * used in xen-api responses.
+     * RFC-3339 date formats can be returned in either Zulu or time zone agnostic.
+     * This list is not an exhaustive list of formats supported by RFC-3339, rather
```

Comment:
> there are one or two formats listed there that are not here 

I've updated the list to contain all applicable ones

> here there are several formats that are not used by the C# SDK [...] shouldn't we be encouraging toolstack to tidy up and standardise the formats

I agree about xapi standardising it - and in fact they're aware and there have been presentation on the matter. Here as a consumer I just want to be safe and not have this regress in the future. There's no testing so we'd only know of future issues if they come in the form of tickets.

In fact, maybe we should improve the C# one too. For that one we have XenCenter so I'm not too worried

Reply:
> In fact, maybe we should improve the C# one too

All languages should use the same formats since they all call the same API. No objections to extending C#, in fact, I would suggest to also audit C and Go since we're here.

Reply:
sure, happy to :) could you flag this PR as needing changes? I'll have a look at the other SDKs too

Reply:
Toolstack currently generates the dates in these two formats, exclusively:
```
20201210T17:19:20
20201210T17:19:20Z
```

Technically in xmlrpc forces us to produce / accept ISO8601 timedates. This is too wide of a format.

The toolstack has a very awkward datetime library, it cannot accept dates with timezones other than Z, for example. And internally it doesn't allow us to use sub-second precision. I'm reworking it and lifting these restrictions. For your interest, this means accepting and producing datetimes with non-Z timezones. Currently I'm using the format `-02:30`, but I can change it if needed.